src/common/schedulers/slurm_commands.py

```diff
@@ -236,7 +236,7 @@ def get_nodes_info(nodes="", command_timeout=DEFAULT_GET_INFO_COMMAND_TIMEOUT):
     show_node_info_command = (
         f'{SCONTROL} show nodes {nodes} | awk \'BEGIN{{RS="\\n\\n" ; ORS="---\\n";}} {{print}}\' | '
         'grep -oP "^NodeName=\\K(\\S+)| NodeAddr=\\K(\\S+)| NodeHostName=\\K(\\S+)| State=\\K(\\S+)|'
-        ' Partitions=\\K(\\S+)|(---)"'
+        ' Partitions=\\K(\\S+)| Reason=\\K((\\S)|(\\s))+ |(---)"'
     )
```

Comment:
Can we use sinfo --json to get this data? (we might need to compare performance difference) Otherwise we need to find a way to make this regex work when a node does not belong to any partition

Reply:
After our discussion, not use sinfo --json and change the regex